icu4c/source/common/uloc.cpp

```diff
@@ -1147,17 +1147,19 @@ uloc_getCurrentLanguageID(const char* oldID){
     return oldID;
 }
 /*
- * the internal functions _getLanguage(), _getCountry(), _getVariant()
+ * the internal functions _getLanguage(), _getScript(), _getRegion(), _getVariant()
  * avoid duplicating code to handle the earlier locale ID pieces
  * in the functions for the later ones by
  * setting the *pEnd pointer to where they stopped parsing
  *
  * TODO try to use this in Locale
  */
-CharString U_EXPORT2
-ulocimp_getLanguage(const char *localeID,
-                    const char **pEnd,
-                    UErrorCode &status) {
+static void
+_getLanguage(const char* localeID,
+             ByteSink* sink,
+             const char** pEnd,
+             UErrorCode& status) {
```

Comment:
optional: Consider *returning* the end pointer rather than writing it to an output argument.

Reply:
I'll keep that in mind, thanks!